lib/main.dart

```diff
@@ -19,7 +19,7 @@ class MyApp extends StatelessWidget {
   Widget build(BuildContext context) {
     // TODO: Implement routing
     return BlocProvider<ManageGeoCubit>(
-      create: (BuildContext context) => ManageGeoCubit()..checkPermission(),
+      create: (BuildContext context) => ManageGeoCubit(),
```

Comment:
Don't you need `.fetchPositionStream()` to be called at this point?

Reply:
That makes more sense than calling it in every widget.  Do you always define cubits for generel purposes globally so that you can use them everywhere?